app_server/test/controllers/UserController.js

```diff
@@ -3,30 +3,59 @@ var Lab = require('lab');
 var lab = exports.lab = Lab.script();
 var Code = require('code');
 
+var Storage = rfr('app/models/Storage.js');
 var Router = rfr('app/Router.js');
+var Facebook = rfr('app/adapters/social_media/Facebook');
+var TestUtils = rfr('test/TestUtils');
+var Service = rfr('app/services/Service');
+
+var testAccount = {userId: 1, username: 'bob', password: 'abc'};
+
+var bob = {
+  username: 'Bob',
+  alias: 'Bob the Builder',
+  email: 'bob@bubblegum.com',
+  password: 'generated',
+  accessToken: 'xyzabc',
+  platformType: 'facebook',
+  platformId: '1238943948',
+  description: 'bam bam bam'
+};
 
 lab.experiment('UserController Tests', function () {
+  lab.beforeEach(function (done) {
+    TestUtils.resetDatabase(done);
+  });
+
   lab.test('Get list of users', function (done) {
-    Router.inject('/api/users', function (res) {
-      Code.expect(res.result).to.equal('Hello Sharmine!');
-      done();
-    });
+    Router.inject({url: '/api/users', credentials: testAccount},
+      function (res) {
+        Code.expect(res.result).to.equal('Hello Sharmine!');
+        done();
+      });
   });
 
   lab.test('Valid id', function (done) {
-    Router.inject('/api/users/213', function (res) {
-      Code.expect(res.result).to.equal('Hello 213!');
-      done();
+    Service.createNewUser(bob).then(function (result) {
+      return Service.getUserById(result.userId);
+    }).then(function(user) {
+      Router.inject({url: '/api/users/' + user.userId,
+                    credentials: testAccount},
+                    function (res) {
+                      Code.expect(res.result.username).to.equal(bob.username);
+                      done();
+                    });
     });
   });
 
   lab.test('Get list of users', function (done) {
-    Router.inject('/api/users/asd', function (res) {
-      var errorMsg = 'child "id" fails because ["id" must be a number]';
-      Code.expect(res.result.statusCode).to.equal(400);
-      Code.expect(res.result.message).to.equal(errorMsg);
-      done();
-    });
+    Router.inject({url: '/api/users/asd', credentials: testAccount},
+      function (res) {
+        var errorMsg = 'child "id" fails because ["id" must be a valid GUID]';
```

Comment:
The name of this test should probably be "Invalid id".
Maybe throw in some symbols instead of "asd"?
And what's a child id?